src/Autofac/RegistrationExtensions.Generics.cs

```diff
@@ -48,5 +51,29 @@ public static partial class RegistrationExtensions
         {
             return OpenGenericRegistrationExtensions.RegisterGeneric(builder, implementer);
         }
+
+        /// <summary>
+        /// Register a delegate that can provide instances of an open generic registration.
+        /// </summary>
+        /// <param name="builder">The container builder.</param>
+        /// <param name="factory">A delegate that receives an array of <see cref="Type"/>, containing the generic type arguments of the requested service.</param>
+        /// <returns>Registration builder allowing the registration to be configured.</returns>
+        public static IRegistrationBuilder<object, OpenGenericDelegateActivatorData, DynamicRegistrationStyle>
+            RegisterGeneric(this ContainerBuilder builder, Func<IComponentContext, Type[], object> factory)
+        {
+            return builder.RegisterGeneric((ctxt, types, parameters) => factory(ctxt, types));
+        }
+
+        /// <summary>
+        /// Register a delegate that can provide instances of an open generic registration.
+        /// </summary>
+        /// <param name="builder">The container builder.</param>
+        /// <param name="factory">A delegate that receives an array of <see cref="Type"/>, containing the generic type arguments of the requested service.</param>
```

Comment:
```suggestion
        /// <param name="factory">Delegate responsible for generating an instance of a closed generic based on the open generic type being registered. It will be called with the current <see cref="IComponentContext"/>, the array of generic type arguments expected to be in the closed generic, and any parameters passed to the resolve operation.</param>
```